### ISSUE TITLE
Evaluate changes to generators for macros / augmentation libraries

### ISSUE THREAD

**Issue by davidmorgan** [dart-lang/sdk#55172]
Macros will introduce augmentation libraries; what changes are needed to widely-used generators so that they continue to work?

The analyzer exposes new API that will need to be used in some cases, see e.g. mentions of "augmentation" in package:analyzer/dart/element/element.dart.

**Comment by davidmorgan** [comment]
Simplest things I could think of with mockito, mocking a macro-added field and a macro-added type, both seem to work without changes

https://github.com/dart-lang/mockito/compare/master...davidmorgan:mockito:built-with-macros

although I did hit an issue with "asset" paths that I'll report over in the "build" repo, and an issue with augmenting a class that generics that I guess I'll add a language test for.

**Comment by eernstg** [comment]
Perhaps this is not a question about language specification documents, but rather about the behavior of code generators? In that case we might consider `area-tools` rather than `area-specification`.

**Comment by davidmorgan** [comment]
Yes, it's not spec related. It's about the effect of macro implementation details on existing code generators ... actually that specifically means the analyzer API, I'll tag it with that. Thanks.
